### DISCUSSION TITLE
Differences between VS2022 and dotnet new files

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#906] [Q&A]
Spinoff of https://github.com/StereoKit/StereoKit/discussions/904 (android geolocation), to figure out some weird things I'm seeing.

I generated a new SK multi-target project from VS2022, and use the following `Program.cs`:

```
using StereoKit;

namespace android_location_test_win
{
    internal class Program
    {
        static void Main(string[] args)
        {
            // Initialize StereoKit
            SKSettings settings = new SKSettings
            {
                appName = "android-location-test-win",
                assetsFolder = "Assets",
            };
            if (!SK.Initialize(settings))
                return;


            // Create assets used by the app
            Pose cubePose = new Pose(0, 0, -0.5f);
            Model cube = Model.FromMesh(
                Mesh.GenerateRoundedCube(Vec3.One * 0.1f, 0.02f),
                Material.UI);

            Matrix floorTransform = Matrix.TS(0, -1.5f, 0, new Vec3(30, 0.1f, 30));
            Material floorMaterial = new Material("floor.hlsl");
            floorMaterial.Transparency = Transparency.Blend;

            Pose windowPose = new Pose(-.2f, 0, -0.6f, Quat.LookDir(0, 0, 1));

            string process_name = "<unset>";

#if ANDROID
            Log.Info($"ProcessName = {Android.App.Application.ProcessName}");
            process_name = Android.App.Application.ProcessName;
#endif

            // Core application loop
            SK.Run(() =>
            {
                if (SK.System.displayType == Display.Opaque)
                    Mesh.Cube.Draw(floorMaterial, floorTransform);

                UI.Handle("Cube", ref cubePose, cube.Bounds);
                cube.Draw(cubePose.ToMatrix());

                UI.WindowBegin("Window", ref windowPose, new Vec2(40, 0) * U.cm, UIWin.Normal);
                UI.Label(process_name);
                UI.WindowEnd();
            });
        }
    }
}
```

This works and shows `ProcessName` on the window, both when I run this from VS2022 on a Quest 3, as well as when I copy the project over to my Linux machine and build it with dotnet 7.0.115, deploy to the Quest and run it there. So this is a known-good configuration. It also confirms that my Linux dotnet environment isn't (completely) broken.

The weird thing is that when I use `dotnet new sk-multi` on Linux to generate a new project, copy in the above `Program.cs` and build and deploy the app fails to run on the Quest 3. Even weirder is that I can no longer uninstall it from the app menu, plus an app icon marked with a blue dot stays around in the system bar, even after a reboot. I seems like the app process gets stuck in some weird state. It's only after a Quest reboot that I can uninstall the app.

When I use `dotnet new sk-multi` on the same **Windows** machine that has VS2022 I see differences in the generated project files. I assumed that having done `dotnet new install StereoKit.Templates` (to 0.7.2) would mean both VS2022 and `dotnet new` (dotnet 7.0.304 in this case) would use the same templates. But I see small differences in, for example, the generated `Platforms/Android/MainActivity.cs`:

![image](https://github.com/StereoKit/StereoKit/assets/1408287/b806c4d8-eb5d-460e-9c35-4e88f6f9eb5b)

![image](https://github.com/StereoKit/StereoKit/assets/1408287/e1bac3a0-76dd-47ea-92c1-212ff53bdd90)

Which makes me wonder if the templates for VS2022 and dotnet (on Windows and Linux) are indeed the same? And if that might cause the differences in app behaviour I'm seeing?

**Comment by maluoi** [comment]
The source template is the same, but the processing applied to it may be a little different for VS2022 vs dotnet? The biggest one I was aware of was that VS2022 will add a solution file! But it does look here like dotnet is also applying a syntax upgrade and document format?

`namespace name {}` is an old style of namespacing, and `namespace name;` is the more modern and recommended format. While I do prefer the more modern style _a lot_, older platforms like UWP don't support it, making me a little hesitant to remove it completely. `dotnet` is likely just applying a fairly trivial syntax transform automatically at start.

Either way, an interesting observation, and now I do kinda wonder what other things might be happening. Maybe the .sln file is forcing a slightly different build toolchain? I'm aware that VS2022's build tools _are_ different, but I'm still not sure how much different, or how/if that would factor into anything on Linux.
